rero_ils/modules/ebooks/dojson/contrib/marc21/model.py

```diff
@@ -162,18 +165,99 @@ def marc21_to_author(self, key, value):
 @marc21.over('title', '^245..')
 @utils.ignore_value
 def marc21_to_title(self, key, value):
-    """Get title.
-
-    title: 245$a
-    without the punctuaction. If there's a $b, then 245$a : $b without the ' /'
+    """Get title data.
+
+    field 245:
+        $a : non repetitive
+        $b : non repetitive
+        $c : non repetitive
+        $n : repetitive
+        $p : repetitive
+        $6 : non repetitive
+    field 246:
+        $a : non repetitive
+        $n : repetitive
+        $p : repetitive
+        $6 : non repetitive
     """
-    main_title = remove_trailing_punctuation(value.get('a'))
-    sub_title = value.get('b')
-    if sub_title:
-        main_title += ' : ' + ' : '.join(
-            utils.force_list(remove_trailing_punctuation(sub_title))
-        )
-    return main_title
+    subfield_245_a = ''
```

Comment:
this transformation is quite long. Can you put some comments along the code to make it more readable.

Reply:
done

Reply:
done